dali/kernels/signal/resampling.h

```diff
@@ -37,25 +40,63 @@ inline double Hann(double x) {
   return 0.5 * (1 + std::cos(x * M_PI));
 }
 
+#ifdef __ARM_NEON
+
+inline float32x4_t vsetq_f32(float x0, float x1, float x2, float x3) {
+    float32x4_t x;
+    x = vdupq_n_f32(x0);
+    x = vsetq_lane_f32(x1, x, 1);
+    x = vsetq_lane_f32(x2, x, 2);
+    x = vsetq_lane_f32(x3, x, 3);
+    return x;
+}
+
+#endif
+
 struct ResamplingWindow {
   inline std::pair<int, int> input_range(float x) const {
-    int i0 = std::ceil(x) - lobes;
-    int i1 = std::floor(x) + lobes;
+    int xc = std::ceil(x);
+    int i0 = xc - lobes;
+    int i1 = xc + lobes;
     return {i0, i1};
   }
 
   inline float operator()(float x) const {
     float fi = x * scale + center;
-    int i = std::floor(fi);
-    float di = fi - i;
+    float floori = std::floor(fi);
+    float di = fi - floori;
+    int i = floori;
     assert(i >= 0 && i < static_cast<int>(lookup.size()));
     return lookup[i] + di * (lookup[i + 1] - lookup[i]);
   }
 
+#ifdef __ARM_NEON
+  inline float32x4_t operator()(float32x4_t x) const {
+    float32x4_t fi = vfmaq_n_f32(vdupq_n_f32(center), x, scale);
+    int32x4_t i = vcvtq_s32_f32(fi);
+    float32x4_t fifloor = vcvtq_f32_s32(i);
+    float32x4_t di = vsubq_f32(fi, fifloor);
+    int idx[4] = {
+        vgetq_lane_s32(i, 0),
+        vgetq_lane_s32(i, 1),
+        vgetq_lane_s32(i, 2),
+        vgetq_lane_s32(i, 3)
+    };
+    float32x2_t c0 = vld1_f32(&lookup[idx[0]]);
+    float32x2_t c1 = vld1_f32(&lookup[idx[1]]);
+    float32x2_t c2 = vld1_f32(&lookup[idx[2]]);
+    float32x2_t c3 = vld1_f32(&lookup[idx[3]]);
+    float32x4x2_t w = vuzpq_f32(vcombine_f32(c0, c1), vcombine_f32(c2, c3));
+    float32x4_t curr = w.val[0];
+    float32x4_t next = w.val[1];
+    return vfmaq_f32(curr, di, vsubq_f32(next, curr));
+  }
+#endif
+
 #ifdef __SSE2__
   inline __m128 operator()(__m128 x) const {
     __m128 fi = _mm_add_ps(x * _mm_set1_ps(scale), _mm_set1_ps(center));
-    __m128i i = _mm_cvtps_epi32(fi);
+    __m128i i = _mm_cvttps_epi32(fi);
```

Comment:
Was doing conversion without truncate an issue before?

Reply:
Yes, it was a bug - I detected it now that there's a slight difference between ARM and SSE and dug deeper to find out that it's SSE that's wrong.